plugins/stats_over_http/stats_over_http.cc

```diff
@@ -574,22 +574,25 @@ stats_origin(TSCont contp ATS_UNUSED, TSEvent event ATS_UNUSED, void *edata)
     goto cleanup;
   }
 
-  int path_len     = 0;
-  const char *path = TSUrlPathGet(reqp, url_loc, &path_len);
+  path = TSUrlPathGet(reqp, url_loc, &path_len);
   TSDebug(PLUGIN_NAME, "Path: %.*s", path_len, path);
 
-  if (!(path_len != 0 && path_len == config->stats_path_len && !memcmp(path, config->stats_path, config->stats_path_len))) {
+  if (!(path_len != 0 && path_len == int(config->stats_path.length()) &&
+        !memcmp(path, config->stats_path.c_str(), config->stats_path.length()))) {
+    TSDebug(PLUGIN_NAME, "not this plugins path, saw: %.*s, looking for: %s", path_len, path, config->stats_path.c_str());
     goto notforme;
   }
 
-  const struct sockaddr *addr = TSHttpTxnClientAddrGet(txnp);
-  if (!is_ip_allowed(config, addr)) {
+  TSSkipRemappingSet(txnp, 1); // not strictly necessary, but speed is everything these days
+
+  addr = TSHttpTxnClientAddrGet(txnp);
```

Comment:
For real C++ style, don't declare this above with an initializer, declare it here initialized by `TSHttpTxnClientAddrGet`.
```
auto addr = TSHttpTxnClienAddrGet(txnp);
```
Not sure why you changed that from the original. If you wanted to be really cutting edge,  you'd do
```
if ( auto addr = TSHttpTxnClientAddrGet(txnp) ; !is _ip_allowed(config, addr)) {
// ....
```